### ISSUE TITLE
**[BUG]** Addon is very unresponsive.

### ISSUE THREAD

**Issue by SexySeamonster** [CGCookie/retopoflow#791]
Retopoflow Version: 3.0.0 Beta

Blender Version/Hash: 2.82

Platform Version/Distribution: Windows PC - Steam

**Issue:**

Addon is very slow to respond to any actions, including UI. Is noticeably faster on a default Blender sphere, but even when remeshing my 250k vert model to roughly 20k verts there was no increase in performance. Tested with a fresh install of Blender with only Retopoflow installed - no difference. 
Attached my blender system-info file and my project Blend file.
PC specs are 4 GHz eight-core processor, 16GB RAM and a GTX 1080 from NVIDIA. 

**How to Reproduce:**

- Download attatched Blend file
- Select any of the LP meshes
- Enter Retopoflow on the corresponding MP mesh
- Try to use the tools (Relax, Tweak) and navigate the options menu

https://drive.google.com/file/d/1DlnSVHSJVD3MxyP30BAQGw7McmGxRP0Z/view?usp=sharing 
[system-info.txt](https://github.com/CGCookie/retopoflow/files/4218031/system-info.txt)

**Comment by vxlcoder** [comment]
Thanks for the report, @SexySeamonster.   Will take a look at this, but might have a few questions for you soon.

**Comment by vxlcoder** [comment]
hmmm... I'm not seeing the same problems when navigating the options menu, but Relax and Tweak certainly are really slow, as well as starting up.

I'm on an 12-core i7-8700 3.20GHz, 32GB RAM, GeForce GTX 1050, Kubuntu.  So, for what we're doing, the differences in our specs shouldn't matter much.  (Note: I also develop on a _much_ less powerful linux laptop, and it performs quite well too)

Let me look into this a bit more.

Questions:

- when you remesh to 20k, are you applying the modifier?
- are you on the very latest 2.82 release?  I've been testing on 2.81 (all versions), 2.82 alpha, 2.82 beta, 2.82 full, 2.83 alpha... and I haven't seen a drop in performance across these, but maybe it's different on Windows?

**Comment by SexySeamonster** [comment]
Yes and yes. By the way, I tested a bit more and when I made a new scene (with same default scene), imported the same exact models and used Retopoflow, it wasn't unresponsive. Well, except for the UI, but that's not as important. No changes in software versions was made.

**Comment by vxlcoder** [comment]
Interesting!  There have been other people report similar issues to you that I simply haven't been able to replicate.  "imported the same exact models ... wasn't unresponsive" gives me a direction to test.  Thanks for working with us on that!

(the UI is still work in progress, but my hunch is that rendering fonts on some machines is much slower than on others)

**Comment by xan2622** [comment]
Hi.

Retopoflow is quite slow for me too. 

On a fresh Blender 2.82 (Official) installation (no other third-party addon installed), after clicking on the **RF: Create new Target** button:
- with a mesh composed of 16,560 faces, it takes 4-5 seconds for the UI to load
- with a mesh composed of 431,340 faces, it takes 15-18 seconds for the UI to load

Note that it ordinary takes 3 seconds for Blender to switch from Object mode to Edit mode with this high poly object.

Other example: with a 5,300 polygons mesh, it takes 4 seconds for the Quick Start Guide to open.

https://drive.google.com/file/d/1dl7QZhN_oqoCmHEtNoafOV33ooOS5I0H/view?usp=sharing

---

**CPU:** Quad Core AMD Phenom II X4 965 3.4 Ghz
**RAM:** 16 GB DDR3
**Graphics card:** NVIDIA GeForce GTX 1070 8 GB (drivers: 442.19)
**Drive:** Samsung SSD 860 EVO 500GB
**OS:** Windows 10 64 Bits

**Comment by vxlcoder** [comment]
@xan2622, your notes are interesting... you mean that Blender itself (not RF) takes 3 seconds to switch from object mode into edit mode?  both of your .blend files load and switch into edit mode quite quickly on my machine.  (RF is still slow, but just discussing Blender's Object -> Edit mode).

i'm leaning toward these slowness issues being a problem with blender not running uniformly across all machines, or different combinations of things cause weird slowness issues in unpredictable ways...

So, just to confirm, your blender takes 3 seconds to switch from obj to edit mode?

**Comment by xan2622** [comment]
> So, just to confirm, your blender takes 3 seconds to switch from obj to edit mode?

Yes, but only with high poly meshes, for example with the _SoccerPlayer2.blend_ file.

Usually, Blender toggles the Object/Edit mode instantly when I press the TAB key.
I haven't really noticed any slowness before (I usually work on scenes with less than 150,000 faces).

I just noticed a real slowness while using Retopoflow, and then, I noticed that (Retopoflow disabled), Blender took 3 seconds to switch from object to edit mode with that high poly mesh.
